umi/sumi/tests/test_demux.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2024 Zero ASIC
+# This code is licensed under Apache License 2.0 (see LICENSE for details)
+
+import pytest
+import multiprocessing
+import numpy as np
+from switchboard import UmiTxRx, random_umi_packet, delete_queue
+
+
+def umi_send(umi_q, n, ports, seed):
+
+    np.random.seed(seed)
+
+    tee = [UmiTxRx(f'tee_{x}.q', '') for x in range(ports)]
+
+    for count in range(n*ports):
+        dstport = np.random.randint(0, ports)
+        dstaddr = (2**8)*np.random.randint(0, 2**32) + (2**(40+dstport))
+        srcaddr = (2**8)*np.random.randint(0, 2**32) + (2**(40+ports))
+        txp = random_umi_packet(dstaddr=dstaddr, srcaddr=srcaddr)
+        print(f"Sending #{count} cmd: 0x{txp.cmd:08x} srcaddr: 0x{srcaddr:08x} "
+              f"dstaddr: 0x{dstaddr:08x} to port {dstport}")
+        # send the packet to both simulation and local queues
+        umi_q.send(txp)
+        tee[dstport].send(txp)
```

Comment:
I feel like this code appears a lot...

Reply:
I refactored the tests and pulled this code out into a common file. Variants of it appeared in demux, mux, and switch. In order to pull the code out, the variants had to be merged into a common format which meant the tests changed slightly as well.